docs/create-pages.mdx

```diff
@@ -68,6 +68,7 @@ const pages = createPages(async ({ createPage, createLayout }) => [
 declare module 'waku/router' {
   interface RouteConfig {
     paths: PathsForPages<typeof pages>;
+    pages: typeof pages;
```

Comment:
So, I'm not sure if you remember our discussion before, but this is going against the abstraction (layers) of waku/router.

#598 "Waku middleware / API stack (v0.20)" may or may not help.

At the `defineRouter` layer, it shouldn't use the notion from `createPages`.
So, `createPages` can depend on `defineRouter`, but not the other way around.

That's why we did `paths: PathsForPages<typeof pages>;` previously.

Reply:
ah, ya sorry. I remember that discussion now. I had a bit of tunnel vision on writing this out, I'll fix that to be similar to how we did for paths, but for the full page types.

If we do something like 
```ts
pages: GetPages<typeof pages>
```
should I then remove `paths: PathsForPages<typeof pages>` since that can then be inferred from the full pages above internally?

This would get us back down to only one item in RouteConfig and also fall in line with the spirit of that last discussion I think

Reply:
I don't prefer using the word or notion of "page", which implies `createPages`.
`paths: PathsForPages<typeof pages>` was really good. defineRouter is lower level of createPages. There can exist another implementation instead of createPages that depends on defineRouter. The difficulty is we don't know what it is yet. But, we would try out best. At least defineRouter is lower level and should be minimal.

`getFoo<typeof foo>` doesn't make sense.

Reply:
I think I might be able to actually just use just the paths that is already there... 🤔

Reply:
ok, updated to extend the existing use of paths internally, but leave the external `paths: PathsForPages` the same

let me know what you think 😄

Reply:
Yeah, looks much better.

Reply:
So, this line should be removed from the doc?

Reply:
yep! sorry, forgot about this, thanks